plugins/hls-alternate-number-format-plugin/src/Ide/Plugin/Literals.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE DerivingVia       #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RankNTypes        #-}
+module Ide.Plugin.Literals where
+import           Data.Maybe                    (fromMaybe, maybeToList)
+import           Data.Set                      (Set)
+import qualified Data.Set                      as S
+import           Data.Text                     (Text)
+import qualified Data.Text                     as T
+import           Development.IDE.GHC.Compat
+import           Development.IDE.GHC.Util      (unsafePrintSDoc)
+import           Development.IDE.Graph.Classes (NFData (rnf))
+import qualified GHC.Generics                  as GHC
+import           Generics.SYB                  (Data, Typeable, cast,
+                                                everything)
+
+-- data type to capture what type of literal we are dealing with
```

Comment:
```suggestion
-- | data type to capture what type of literal we are dealing with
```

Make it haddock :)

Reply:
Updated (changed the wording to be less clunky). That was mostly an internal comment.